### PR TITLE
Adding ARM image overrides for metal IPI.

### PR DESCRIPTION
**Pull request by tvardema**
JIRA: [ARMOCP-384](https://issues.redhat.com//browse/ARMOCP-384)
CC: @aleskandro

### CODE REVIEW

File: ci-operator/step-registry/baremetalds/devscripts/proxy/baremetalds-devscripts-proxy-commands.sh [hunk 1]
```diff
@@ -37,12 +37,19 @@ sudo systemctl start firewalld
 sudo firewall-cmd --add-port=8213/tcp --permanent
 sudo firewall-cmd --reload
 
+
+SQUID_IMAGE=quay.io/sameersbn/squid:latest
+if [ "${ARCHITECTURE}" == "arm64" ]; then
+  # Test image reference, will update at a later date
+  SQUID_IMAGE=quay.io/openshifttest/squid-proxy:multiarch
+fi
```
Comment (yselkowitz) [inline ci-operator/step-registry/baremetalds/devscripts/proxy/baremetalds-devscripts-proxy-commands.sh:45]: why not just use the `openshifttest` image on all arches?

File: ci-operator/step-registry/baremetalds/devscripts/proxy/baremetalds-devscripts-proxy-commands.sh [hunk 2]
```diff
@@ -37,12 +37,19 @@ sudo systemctl start firewalld
...
+
 sudo podman run -d --rm \
      --net host \
      --volume \$HOME/squid.conf:/etc/squid/squid.conf \
      --name external-squid \
      --dns 127.0.0.1 \
-     quay.io/sameersbn/squid:latest
+     \$SQUID_IMAGE
```
Comment (yselkowitz) [inline ci-operator/step-registry/baremetalds/devscripts/proxy/baremetalds-devscripts-proxy-commands.sh:52]: Ditto.

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh [hunk 3]
```diff
@@ -95,25 +95,35 @@ mkdir dev-scripts
...
 
-echo "export OPENSHIFT_RELEASE_IMAGE=${OPENSHIFT_INSTALL_RELEASE_IMAGE}" >> /root/dev-scripts/config_root.sh
 echo "export ADDN_DNS=\$(awk '/nameserver/ { print \$2;exit; }' /etc/resolv.conf)" >> /root/dev-scripts/config_root.sh
 echo "export OPENSHIFT_CI=true" >> /root/dev-scripts/config_root.sh
 echo "export NUM_WORKERS=3" >> /root/dev-scripts/config_root.sh
 echo "export WORKER_MEMORY=16384" >> /root/dev-scripts/config_root.sh
 echo "export ENABLE_LOCAL_REGISTRY=true" >> /root/dev-scripts/config_root.sh
 
+if [[ "${ARCHITECTURE}" == "arm64" ]]; then
+  echo "export DOCKER_REGISTRY_IMAGE=quay.io/libpod/registry:2.8" >> /root/dev-scripts/config_root.sh
```
Comment (yselkowitz) [inline ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh:116]: What is this overriding, and why only on ARM?

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh [hunk 4]
```diff
@@ -95,25 +95,35 @@ mkdir dev-scripts
...
+  echo "export OPENSHIFT_RELEASE_IMAGE=${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}" >> /root/dev-scripts/config_root.sh
+  echo "export IRONIC_LOCAL_IMAGE=\\\$(oc adm release info -a /root/dev-scripts/pull_secret.json \
+    \\\${OPENSHIFT_RELEASE_IMAGE} -o json | jq -r '.references.spec.tags[] | select(.name==\"ironic\").from.name')" \
+    >> /root/dev-scripts/config_root.sh
+  echo "export SUSHY_TOOLS_IMAGE=quay.io/multi-arch/sushy-tools:muiltarch" >> /root/dev-scripts/config_root.sh
+  echo "export VBMC_IMAGE=quay.io/multi-arch/vbmc:arm" >> /root/dev-scripts/config_root.sh
```
Comment (yselkowitz) [inline ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh:122]: We should find better locations for these.

Comment (tvardema) [reply ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh:122]: After an offline conversation, the `quay.io/multi-arch` registry is not considered a permanent solution, so these values should be updated with something more appropriate.

Comment (jeffdyoung) [reply ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh:122]: I had discussed building these with some of the BM folks a while ago. The blocker was that they use [quay](https://quay.io/organization/metal3-io) for builds, and quay don't support arm builds ([yet](https://issues.redhat.com/browse/PROJQUAY-3059)). As a work around, I manually these to test. Happy to discuss more appropriate options.

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh (hunk 3 above)
Comment (jeffdyoung) [reply ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh:116]: Went ahead and opened [this](https://github.com/openshift-metal3/dev-scripts/pull/1428)

The image in question will work for the architectures we care about.

File: ci-operator/step-registry/baremetalds/devscripts/proxy/baremetalds-devscripts-proxy-commands.sh (hunk 1 above)
Comment (deepsm007) [reply ci-operator/step-registry/baremetalds/devscripts/proxy/baremetalds-devscripts-proxy-commands.sh:45]: We would have to test `openshifttest` image

Comment (tvardema) [reply ci-operator/step-registry/baremetalds/devscripts/proxy/baremetalds-devscripts-proxy-commands.sh]: Testing now.

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh (hunk 3 above)
Comment (tvardema) [reply ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh:116]: I'm slightly confused on this comment... Is the suggestion to move the overwrite to all arches, or is this enough clarification that we'll leave it as is?

File: ci-operator/step-registry/baremetalds/devscripts/proxy/baremetalds-devscripts-proxy-commands.sh (hunk 1 above)
Comment (tvardema) [reply ci-operator/step-registry/baremetalds/devscripts/proxy/baremetalds-devscripts-proxy-commands.sh]: I updated this section to include the multiarch image, and it seems to not cause any problems.  Will resolve the thread here.

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh (hunk 3 above)
Comment (tvardema) [reply ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh]: A PR was merged that makes this specific line unnecessary (exporting a new DOCKER_REGISTRY_IMAGE I mean)

File: ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh [hunk 5]
```diff
@@ -230,6 +230,10 @@ do
 
   count=$((count+1))
   if (( count > 20 )); then
+#    echo "#### LOOK HERE ####"
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh:233]: hey @tvardema should we get rid of this?

File: ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh [hunk 6]
```diff
@@ -94,7 +94,13 @@ cat > packet-config.yaml <<-EOF
 EOF
 ansible-playbook packet-config.yaml |& gawk '{ print strftime("%Y-%m-%d %H:%M:%S"), $0; fflush(); }'
 
+#if [ -e "${CLUSTER_PROFILE_DIR}/ofcir_url" ] && [ "${ARCHITECTURE}" != "arm64" ] ; then
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh:97]: ^

Comment (tvardema) [reply ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh:97]: Yes we should remove the comments.  Will resubmit shortly.

**Comment by andfasano** [comment]
Moving to @derekhiggins, right now actively maintaining the metal jobs

/assign @derekhiggins

**Review by deepsm007 — APPROVED** [review]

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml [hunk 7]
```diff
@@ -8,6 +8,15 @@ ref:
       cpu: 1000m
       memory: 500Mi
   env:
+  - name: ARCHITECTURE
+    default: ""
+    documentation: Set the architecture for ARM image replacement
+  - name: PACKET_PLAN
```
Comment (derekhiggins) [inline ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml:14]: This isn't being used, can it be removed?

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml [hunk 8]
```diff
@@ -17,5 +26,7 @@ ref:
   dependencies:
   - name: "release:latest"
     env: OPENSHIFT_INSTALL_RELEASE_IMAGE
+  - name: "release:latest"
+    env: OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE
```
Comment (derekhiggins) [inline ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml:30]: Is there a reason not to just reuse OPENSHIFT_INSTALL_RELEASE_IMAGE ?

File: ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.11.yaml [hunk 9]
```diff
@@ -157,6 +157,19 @@ tests:
       BASE_DOMAIN: arm-ci.devcluster.openshift.com
       COMPUTE_NODE_TYPE: m6g.xlarge
     workflow: openshift-e2e-aws
+- as: ocp-e2e-metal-ipi-ovn-arm64
+  cluster: build05
```
[the remark this answers is not in the document]
Comment (deepsm007) [reply ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.11.yaml:161]: We can remove the `cluster: build05` and let CI pick any build cluster

File: ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.11.yaml [hunk 10]
```diff
@@ -157,6 +157,19 @@ tests:
...
+  interval: 72h
+  steps:
+    cluster_profile: equinix-ocp-metal
```
[the remark this answers is not in the document]
Comment (deepsm007) [reply ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.11.yaml:164]: Yes, it is intended as periodic and no additional usage

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml (hunk 8 above)
Comment (deepsm007) [reply ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml:30]: The build cluster requires cli and other functionalities specific to arch on which ci-operator runs on. Above configuration allows the cli to be used from x86 payload (release:latest) and only overrides the installation with arm payload.

File: ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.11.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (tvardema) [reply ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.11.yaml:161]: Done

File: ci-operator/step-registry/baremetalds/devscripts/proxy/baremetalds-devscripts-proxy-ref.yaml [hunk 11]
```diff
@@ -8,6 +8,9 @@ ref:
       cpu: 1000m
       memory: 500Mi
   env:
+  - name: ARCHITECTURE
```
[the remark this answers is not in the document]
Comment (tvardema) [reply ci-operator/step-registry/baremetalds/devscripts/proxy/baremetalds-devscripts-proxy-ref.yaml:11]: Done

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh [hunk 12]
```diff
@@ -98,25 +98,36 @@ mkdir dev-scripts
...
 echo "export NUM_WORKERS=3" >> /root/dev-scripts/config_root.sh
 echo "export WORKER_MEMORY=16384" >> /root/dev-scripts/config_root.sh
 echo "export ENABLE_LOCAL_REGISTRY=true" >> /root/dev-scripts/config_root.sh
 
+if [[ "${ARCHITECTURE}" == "arm64" ]]; then
+  echo "export OPENSHIFT_RELEASE_IMAGE=${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}" >> /root/dev-scripts/config_root.sh
+  #echo "export IRONIC_LOCAL_IMAGE=\\\$(oc adm release info -a /root/dev-scripts/pull_secret.json \
+  #  \\\${OPENSHIFT_RELEASE_IMAGE} --image-for=\"ironic\")" >> /root/dev-scripts/config_root.sh
+  echo "export IRONIC_IMAGE=\\\$(oc adm release info -a /root/dev-scripts/pull_secret.json \
+    \\\${OPENSHIFT_RELEASE_IMAGE} --image-for=\"ironic\")" >> /root/dev-scripts/config_root.sh
```
[the remark this answers is not in the document]
Comment (tvardema) [reply ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh:123]: I'll make a comment to look into a change in dev-scripts, and then we can remove this change.

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml (hunk 7 above)
Comment (tvardema) [reply ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml:14]: Done

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml (hunk 8 above)
Comment (derekhiggins) [reply ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml:27]: ahh, makes sense, thanks.

**Pull request merged**